src/btree/bt_page.c

```diff
@@ -416,7 +416,7 @@ __wt_page_inmem(WT_SESSION_IMPL *session, WT_REF *ref, const void *image, uint32
      * used to instantiate the page image even though the values might not match exactly, because
      * that's the only value we have when discarding the page image and accounting needs to match.
      */
-    size = LF_ISSET(WT_PAGE_DISK_ALLOC) ? dsk->mem_size : 0;
+    size = LF_ISSET(WT_PAGE_DISK_ALLOC) ? page->dsk_alloc_size = image_alloc_size : 0;
```

Comment:
Is this right? It will result in `size = page->dsk_alloc_size = image_alloc_size;` It's weird to do that in a ternary (cascading assignments is something I need to think about anyway, putting them in a ternary makes me wonder if it is deliberate).

Reply:
It appears correct to me. If the `WT_PAGE_DISK_ALLOC` flag is not set, we intend to maintain `page->dsk_alloc_size` as 0 without altering it. We would only modify `page->dsk_alloc_size` and track the `size` if `WT_PAGE_DISK_ALLOC` is set.

If readability is your concern I am happy to change it.